subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -1,17 +1,43 @@
 /*
  * Copyright (c) 2018-2021 mcumgr authors
+ * Copyright (c) 2022 Laird Connectivity
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <assert.h>
+#include <zephyr.h>
 #include <limits.h>
 #include <string.h>
 #include <sys/util.h>
+#include <fs/fs.h>
 #include "cborattr/cborattr.h"
 #include "mgmt/mgmt.h"
 #include "fs_mgmt/fs_mgmt.h"
 #include "fs_mgmt/fs_mgmt_impl.h"
 #include "fs_mgmt/fs_mgmt_config.h"
+#include "fs_mgmt/hash_checksum_mgmt.h"
+
+#if defined(CONFIG_FS_MGMT_CHECKSUM_IEEE_CRC32)
+#include "fs_mgmt/hash_checksum_crc32.h"
+#endif
+
+#if defined(CONFIG_FS_MGMT_HASH_SHA256)
+#include "fs_mgmt/hash_checksum_sha256.h"
+#endif
+
+#ifdef CONFIG_FS_MGMT_CHECKSUM_HASH
+BUILD_ASSERT(sizeof(CONFIG_FS_MGMT_CHECKSUM_HASH_DEFAULT) > 1,
+	     "Missing mcumgr fs checksum/hash algorithm selection?");
+BUILD_ASSERT(CONFIG_FS_MGMT_CHECKSUM_HASH_LARGEST_OUTPUT_SIZE > 0,
+	     "Missing mcumgr fs checksum/hash algorithm selection?");
```

Comment:
I am not sure whether this is needed; an user will not be able to set either of these with proj.conf because it is not allowed to set Kconfig options without prompt.

Reply:
They can enable the feature and not enable any algorithms for it which this catches